redi-connect-front/src/pages/front/signup/signUpFormFactory.tsx

```diff
@@ -111,210 +111,224 @@ export const buildSignUpForm = (
-                props.handleChange,
-                props.setFieldTouched
-              )}
-              fullWidth
-              margin="normal"
-            />
-            <TextField
-              id="passwordConfirm"
-              name="passwordConfirm"
-              type="password"
-              helperText={
-                props.touched.passwordConfirm
-                  ? props.errors.passwordConfirm
-                  : ''
-              }
-              error={
-                props.touched.passwordConfirm &&
-                Boolean(props.errors.passwordConfirm)
-              }
-              label="Repeat password*"
-              value={props.values.passwordConfirm}
-              onChange={change.bind(
-                null,
-                'passwordConfirm',
-                props.handleChange,
-                props.setFieldTouched
-              )}
-              fullWidth
-              margin="normal"
-            />
-            {type === 'public-sign-up-mentee-pending-review' && (
-              <FormControl fullWidth>
-                <InputLabel htmlFor="mentee_currentlyEnrolledInCourse">
+    <Columns vCentered>
+      <Columns.Column
+        size={6}
+        responsive={{ mobile: { hide: { value: true } } }}
+      >
+        <SignUp />
+      </Columns.Column>
+
+      <Columns.Column size={5} offset={1}>
+        <form onSubmit={e => e.preventDefault()}>
+          <Heading
+            size={1}
+            weight="normal"
+            renderAs="h1"
+            className="title--border"
+          >
+            Sign-up
+          </Heading>
+          <Form.Field>
+            <Form.Control>
+              <Form.Input
+                id="firstName"
+                name="firstName"
+                color={
+                  formik.touched.firstName && Boolean(formik.errors.firstName)
+                    ? "danger"
+                    : null
+                }
+                placeholder="First name*"
+                value={formik.values.firstName}
+                onChange={onChange.bind(null, "firstName")}
+              />
+            </Form.Control>
+          </Form.Field>
+          <Form.Field>
+            <Form.Control>
+              <Form.Input
+                id="lastName"
+                name="lastName"
+                color={
+                  formik.touched.lastName && Boolean(formik.errors.lastName)
+                    ? "danger"
+                    : null
+                }
+                placeholder="Last name*"
+                value={formik.values.lastName}
+                onChange={onChange.bind(null, "lastName")}
+              />
+            </Form.Control>
+          </Form.Field>
+          <Form.Field>
+            <Form.Control>
+              <Form.Input
+                id="username"
+                name="username"
+                type="email"
+                color={
+                  formik.touched.username && Boolean(formik.errors.username)
```

Comment:
Further to my previous comment about creating a `FormInput` component, you could also here create a simple helper function:
`const determineInputColor = (field: String) => (formik.touched[field] && Boolean(formik.errors[field])) ? "danger" : null`
then its use is is simple: `color={determineInputColor('username')}`

Reply:
makes sense, I'm still not sure if this is the best solution for validation of formik. That is the reason why I kept it inline.

Reply:
Btw. I remember trying the same thing you are proposing and then TypeScript came between. It's not working btw, so I left it for later consideration...

Reply:
Ah, I see the TS issue. Just do this instead:
`const determineInputColor = (field: keyof SignUpFormValues) => (formik.touched[field] && Boolean(formik.errors[field])) ? "danger" : null`

... and the result is pure TS candy - look at how the intellisense works for me now:

![Screenshot 2020-04-03 at 12 23 48](https://user-images.githubusercontent.com/10571311/78350924-47551280-75a6-11ea-9b97-f323e1662a5a.png)

Reply:
huh how ugly TS code is...

Reply:
anyway, just added the candy to the fix package

Reply:
🤣